### PR TITLE
Execution improvements and peer/job configuration changes.

### PR DESCRIPTION
**Pull request by jasebell**

### CODE REVIEW

**Comment by otfrom** [comment]
fix the PR title

File: src/kixi/hecuba/onyx/jobs/measurements_connector.clj [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+(defn push-payload-to-hecuba
+  "Create the http post request for measurements
+  uploads"
+  [json-payload entity-id device-id]
+  (let [json-to-send (json/generate-string {:measurements json-payload})
+        endpoint (str (env :hecuba-endpoint) "entities/" entity-id "/devices/" device-id "/measurements/")]
+
+    (timbre/info (format "DATA: %s" json-to-send))
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/jobs/measurements_connector.clj]: Use timbre/infof ;)

File: src/kixi/hecuba/onyx/jobs/measurements_connector.clj [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    (timbre/info "Using endpoint: %s" (str (env :hecuba-endpoint) "entities/" entity-id "/devices/" device-id "/measurements/"))
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/jobs/measurements_connector.clj]: This looks wrong.. no format call? infof to the rescue!

File: src/kixi/hecuba/onyx/weather.clj [hunk 3]
```diff
@@ -49,7 +55,7 @@
        (clojure.string/join \newline errors)))
 
 (defn exit [status msg]
-  (println msg)
+  (timbre/info msg)
   (System/exit status))
 
 (defn assert-job-exists [job-name]
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/weather.clj:61]: Didn't we work out that this doesn't work at all?

File: src/kixi/hecuba/onyx/weather.clj [hunk 4]
```diff
@@ -58,30 +64,30 @@
       (error-msg (into [(str "There is no job registered under the name " job-name "\n")
                         "Available jobs: "] (keys jobs))))))
 
+(defn start-job [job-name peer-config options]
+  (let [job-id (:job-id
+                (onyx.api/submit-job peer-config
+                                     (onyx.job/register-job job-name (:config options))))]
+    (timbre/info (format "Successfully submitted job: %s (%s)" job-id job-name))))
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/weather.clj]: infof

File: src/kixi/hecuba/onyx/weather.clj [hunk 5]
```diff
@@ -58,30 +64,30 @@
...
-      (assert-job-exists job-name)
-      (let [job-id (:job-id
-                    (onyx.api/submit-job peer-config
-                                         (onyx.job/register-job job-name config)))]
-        (println "Successfully submitted job: " job-id)
-        (println "Blocking on job completion...")
-        (onyx.test-helper/feedback-exception! peer-config job-id)
-        (onyx.api/await-job-completion peer-config job-id)
-        (exit 0 "Job Completed")))))
+          active-jobs (set (map first (methods onyx.job/register-job)))]
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/weather.clj]: active-jobs unused?

File: src/kixi/hecuba/onyx/jobs/weather_connector.clj [hunk 6]
```diff
@@ -71,15 +71,15 @@
 (defn get-met-office-csv-for-day-hour
   "Get Metoffice data as a string"
   [querydate querytime siteid]
-  (timbre/info "get-met-office-csv-for-day-hour")
+  (timbre/info (format "get-met-office-csv-for-day-hour qdate:%s qtime:%s siteid:%s" querydate querytime siteid))
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/jobs/weather_connector.clj]: infof

File: src/kixi/hecuba/onyx/jobs/weather_connector.clj [hunk 7]
```diff
@@ -71,15 +71,15 @@
...
   (->> (met-office-post-request querydate querytime siteid)
        (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
        (client/get)
        :body))
 
 
 (defn pull-weather-station-day-data [querydate siteid]
-  (timbre/info "pull-weather-station-day-data")
+  (timbre/info (format "pull-weather-station-day-data siteid:%s querydate:%s" siteid querydate) )
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/jobs/weather_connector.clj]: infof

File: src/kixi/hecuba/onyx/jobs/measurements.clj [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+(def workflow
+  [[:event/in-queue       :event/save-measurements]
+   [:event/save-measurements  :event/out-confirm]])
+
+(def kafka-consumer-task-opts
+  {:onyx/min-peers 1 ;; should be number of partitions
+   :onyx/max-peers 1
+   :kafka/topic "hecuba-measurements-queue"
+   :kafka/group-id "kixi-hecuba-weather"
+   :kafka/zookeeper "127.0.0.1:2181"
```
Comment (tcoupland) [inline src/kixi/hecuba/onyx/jobs/measurements.clj]: should this be zookeeper like the others?

File: src/kixi/hecuba/onyx/weather.clj (hunk 3 above)
Comment (jasebell) [inline src/kixi/hecuba/onyx/weather.clj:61]: No, but the exit function is used in a couple of other places so I'd rather leave it in.

**Comment by tcoupland** [comment]
:shipit:

**Pull request merged**